source/ide/ide_methods.bas

```diff
@@ -1312,6 +1306,8 @@ FUNCTION ide2 (ignore)
         exitvalue = _EXIT
         IF (exitvalue AND 1) <> 0 OR ideexit <> 0 THEN ideexit = 0: GOTO quickexit
 
+        IF _FILEEXISTS(RecentFile$ + FlagExt$) THEN SyncHistory "RECENT" 'recent file list changed, sync required

```

Comment:
Flag/signal file existence just triggers the sync, it's content otherwise is used by instances to  login/logout themself form sync, so the file is required.

Reply:
The flag file is only necessary to signal that the recent or history file has changed, correct? The mtime can tell you that without having to continually open/read the flag file, so it's significantly cheaper.

I guess separately, I'd be concerned that the instUseCnt will end up wrong when someone kills QB64 via TaskManager (or it crashes) rather than a clean exit.

Reply:
This is in the main IDE loop, isn't it? If so, then the IDE will open and read that sync file ~100 times a second, that's my concern.

Reply:
Do you really think I wouldn't have checked for that impact before implementing it that way. A loop of 100000 OPEN/GET/PUT/CLOSE operations took ~13 seconds on my 14 years old laptop with mechanical harddrive. That's means an average of somewhat more than a 100th second for that 100 loops.

Reply:
If you check the `mtime` then the whole issue goes away, you don't have to have that risk of slowdown or have to deal with the flag file at all.

Based on your comment about `AddToHistory` you probably wouldn't need the locking either, since two instances will never be writing to the history file at the same time.